Cython/Compiler/UtilNodes.py

```diff
@@ -291,6 +291,42 @@ def generate_evaluation_code(self, code):
         self.subexpression.generate_evaluation_code(code)
         self.teardown_temp_expr(code)
 
+# distinguish the comprehension classes just so they can be
+# identified separately
+class ComprehensionEvalWithTempExprNode(EvalWithTempExprNode):
+    def setup_temp_expr(self, code):
```

Comment:
This whole method looks like a collection of things that other nodes and transformations are already taking care of. It seems dangerous (in terms of future maintenance) to replicate them here.
That (again) suggests to me that these input values should be much less special overall.